### ISSUE TITLE
VSCode Dev Containers processes do not receive SIG signals

### ISSUE THREAD

**Issue by anthonyma94** [microsoft/vscode-remote-release#8257]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.76.2
- OS Version: WSL Ubuntu 22.04 with Docker Desktop

When running a process inside a dev container, the container does not emit any signals when terminating a process. E.g. if I have a Node.JS app and I run it with `npm start`, then I terminate the process via Ctrl+C, SIGINT or SIGTERM are not sent. Same thing if I create a VSCode task and then terminate said task.

Steps to Reproduce:

1. Create a new Node dev container
2. Run `npm start`
3. Try exiting

**Comment by anthonyma94** [comment]
No, I mean the interrupt signals aren't passed to node if I have a listener.

```ts
["SIGINT", "SIGTERM", "SIGUSR1", "SIGUSR2"].forEach(sig => {
  process.on(sig, () => {
    console.log(`Received ${sig}`) // not logged
    process.exit()
  }
}
```

This code runs when running bare metal, or built into a production container. In dev container the process exits without any logs. The exception is when I have a watcher set up (ts-node reload on save) and then update a file, SIGTERM is received by the process.

**Comment by chrmarti** [comment]
Are you saying it works when you run it inside `docker run` of a non-dev container? Does it work when run inside `docker exec` on a non-dev container?

**Comment by anthonyma94** [comment]
It works with `docker run` of a non-dev container. No idea how I could run it with `docker exec`, could you give a few pointers?

**Comment by chrmarti** [comment]
You could do `docker run -d <image> sleep inf` and use the container id that returns for `docker exec -it <container id> bash` to get a shell in the container using `exec`. (Please check if I have the command arguments correct.)

**Comment by anthonyma94** [comment]
Can confirm it also works when run inside `docker exec` on a non-dev container.
